pyteal/ast/abi/address.py

```diff
@@ -95,7 +99,10 @@ def set(
                     f"Got bytes with length {len(value)}, expected {AddressLength.Bytes}"
                 )
             case Expr():
-                return self.stored_value.store(value)
+                return Seq(
+                    Assert(Len(value) == Int(AddressLength.Bytes.value)),
+                    self.stored_value.store(value),
+                )
             case CollectionSequence():
```

Comment:
should we check the length of this as well?

Reply:
hole up let me see... I think this check can be done on python level, right?

Namely, `len(value) == AddressLength.Bytes.value`?

Reply:
nvm, adding a few lines of checks